test/uk/gov/hmrc/excisemovementcontrolsystemapi/filters/MovementFilterSpec.scala

```diff
@@ -1,123 +0,0 @@
-/*
- * Copyright 2023 HM Revenue & Customs
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package uk.gov.hmrc.excisemovementcontrolsystemapi.filters
-
-import org.scalatestplus.play.PlaySpec
-import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.Movement
-
-import java.time.Instant
-
-class MovementFilterSpec extends PlaySpec {
```

Comment:
If we add tests for those filters(`arc`, `lrn`, `updatedSince`) aren't we then just testing if Mongo's queries are working properly ? Only case that I see valid testing there is one for `traderType` because it has some logic around it.

Reply:
What we are testing is that if you have one filter set it will filter the data in the Db for that filter, you could then test multiple and make sure they are all "and" together. Youc an then test with an empty filter and make sure that you get back ones for the consignor or consginee, so I would disagree that we are only testing the mongo filters as we're testing all the logic in the getMovementByErn method in the repository. These tests would have given you an error when you accidentally used eq instead of gte too

Reply:
basically replicating these tests but in the MovementRepositoryITSpec and having data added to the db, then filtering